msalautomationapp/src/androidTest/java/com/microsoft/identity/client/msal/automationapp/testpass/local/TestCasePerf.java

```diff
@@ -0,0 +1,164 @@
+import java.util.Arrays;
+
+// Silent Auth with force_refresh
+// https://identitydivision.visualstudio.com/DefaultCollection/IDDP/_workitems/edit/99563
+public class TestCasePerf extends AbstractMsalUiTest {
+
+    @Test
+    public void test_99563() {
+        final TokenRequestLatch latch = new TokenRequestLatch(1);
+
+        final AcquireTokenParameters parameters = new AcquireTokenParameters.Builder()
+                .startAuthorizationFromActivity(mActivity)
+                .withLoginHint(mLoginHint)
+                .withScopes(Arrays.asList(mScopes))
+                .withCallback(successfulInteractiveCallback(latch))
+                .withPrompt(Prompt.SELECT_ACCOUNT)
+                .build();
+
+
+        final InteractiveRequest interactiveRequest = new InteractiveRequest(
+                mApplication,
+                parameters,
+                new OnInteractionRequired() {
+                    @Override
+                    public void handleUserInteraction() {
+                        final String username = mLoginHint;
+                        final String password = LabConfig.getCurrentLabConfig().getLabUserPassword();
+
+                        final PromptHandlerParameters promptHandlerParameters = PromptHandlerParameters.builder()
+                                .prompt(PromptParameter.SELECT_ACCOUNT)
+                                .loginHint(mLoginHint)
+                                .sessionExpected(false)
+                                .consentPageExpected(false)
+                                .speedBumpExpected(false)
+                                .build();
+
+                        new AadPromptHandler(promptHandlerParameters)
+                                .handlePrompt(username, password);
+                    }
+                }
+        );
+
+        interactiveRequest.execute();
+        latch.await(TokenRequestTimeout.SHORT);
+
+        final IAccount account = getAccount();
+
+        CodeMarkerManager.setEnableCodeMarker(true);
+        final int numberOfOccurrenceOfTest = 10;
+        final String outputFilenamePrefix = "PerfDataTarget";
+        for(int i = 0; i < numberOfOccurrenceOfTest; i++) {
+            CodeMarkerManager.clear();
+            final TokenRequestLatch silentLatch = new TokenRequestLatch(1);
+
+            final AcquireTokenSilentParameters silentParameters = new AcquireTokenSilentParameters.Builder()
+                    .forAccount(account)
+                    .fromAuthority(account.getAuthority())
+                    .forceRefresh(true)
+                    .withScopes(Arrays.asList(mScopes))
+                    .withCallback(successfulSilentCallback(silentLatch))
+                    .build();
+
+            mApplication.acquireTokenSilentAsync(silentParameters);
+            silentLatch.await(TokenRequestTimeout.SILENT);
+            try {
+                FileAppender fileAppender = new FileAppender(outputFilenamePrefix + i + ".txt", new SimpleTextFormatter());
```

Comment:
I see so this is not being used for logging but just for the purpose of writing to a file...is that right?

Reply:
Yes that's write. It is just for the purpose of writing to the PerfData text file. I have re-used the FileAppender and created a new SimpleTextFormatter for this.